src/modules/iotjs_module_dbus.c

```diff
@@ -280,12 +280,24 @@ static DBusHandlerResult iotjs_dbus_handle_message(DBusConnection* conn,
   jerry_set_object_native_pointer(jmsg, msg, &this_module_native_info);
 
   iotjs_jargs_t jargs = iotjs_jargs_create(5);
-  iotjs_jargs_append_jval(&jargs, jerry_create_string(sender));
-  iotjs_jargs_append_jval(&jargs, jerry_create_string(object_path));
-  iotjs_jargs_append_jval(&jargs, jerry_create_string(interface));
-  iotjs_jargs_append_jval(&jargs, jerry_create_string(member));
+  jerry_value_t jsender = jerry_create_string(sender);
+  jerry_value_t jobject_path = jerry_create_string(object_path);
+  jerry_value_t jinterface = jerry_create_string(interface);
+  jerry_value_t jmember = jerry_create_string(member);
+  iotjs_jargs_append_jval(&jargs, jsender);
+  iotjs_jargs_append_jval(&jargs, jobject_path);
+  iotjs_jargs_append_jval(&jargs, jinterface);
+  iotjs_jargs_append_jval(&jargs, jmember);
   iotjs_jargs_append_jval(&jargs, jmsg);
-  iotjs_make_callback(_this->jcallback, jerry_create_undefined(), &jargs);
+  jerry_value_t jmethod_this = jerry_create_undefined();
+  iotjs_make_callback(_this->jcallback, jmethod_this, &jargs);
+  jerry_release_value(jmethod_this);
+  jerry_release_value(jsender);
+  jerry_release_value(jobject_path);
+  jerry_release_value(jinterface);
+  jerry_release_value(jmember);
+  jerry_release_value(jmsg);
+  iotjs_jargs_destroy(&jargs);
```

Comment:
This looks really complicated, could we use macros to beautify them?

Reply:
it's hard to use macros to manage variable's lifecycle in jerry and that's why i opened #186, using data block is one of the resolution to resolve this complicated question,

Reply:
Could you please try with JerryScript's experimental autorelease feature? See the [autorelease.h](https://github.com/Rokid/ShadowNode/blob/master/deps/jerry/jerry-ext/include/jerryscript-ext/autorelease.h).